src/app/dao/tests/PartDAO.spec.ts

```diff
@@ -60,43 +84,363 @@ describe('PartDAO', () => {
+            const result: boolean = await partDAO.userHasActivePart(creator);
             // Then it should return false
             expect(result).toBeFalse();
 
         });
         afterEach(async() => {
-            await signOut(TestBed.inject(Auth));
+            await signOut();
+        });
+    });
+    describe('for verified user', () => {
+        it('should allow creating a part', async() => {
+            // Given a verified user
+            const creator: MinimalUser = await createConnectedUser(CREATOR_EMAIL, CREATOR_NAME);
+            // When creating a part
+            const result: Promise<string> = partDAO.create({ ...PartMocks.INITIAL, playerZero: creator });
+            // Then it should succeed
+            await expectAsync(result).toBeResolved();
+        });
+        it('should allow reading parts', async() => {
+            // Given a part and a verified user
+            const creator: MinimalUser = await createConnectedUser(CREATOR_EMAIL, CREATOR_NAME);
+            const part: Part = { ...PartMocks.INITIAL, playerZero: creator };
+            const partId: string = await partDAO.create(part);
+            await signOut();
+
+            await createConnectedUser(MALICIOUS_EMAIL, MALICIOUS_NAME);
+            // When reading the part
+            const result: Promise<MGPOptional<Part>> = partDAO.read(partId);
+            // Then it should succeed
+            await expectAsync(result).toBeResolvedTo(MGPOptional.of(part));
+        });
+        it('should forbid non-player to change fields', async() => {
+            // Given a part, and a user who is not playing in this game
+            const creator: MinimalUser = await createConnectedUser(CREATOR_EMAIL, CREATOR_NAME);
+            const partId: string = await partDAO.create({ ...PartMocks.INITIAL, playerZero: creator });
+            await signOut();
+            await createConnectedUser(MALICIOUS_EMAIL, MALICIOUS_NAME);
+
+            const updates: Partial<Part>[] = [
+                { lastUpdate: { index: 1, player: 0 } },
+                { turn: 42 },
+                { result: 3 },
+                { listMoves: [{ a: 1 }] },
+                { lastUpdateTime: serverTimestamp() },
+                { remainingMsForZero: 42 },
+                { remainingMsForOne: 42 },
+                { winner: 'me!' },
+                { loser: 'you' },
+                { scorePlayerZero: 42 },
+                { scorePlayerOne: 42 },
+                { request: Request.rematchProposed(Player.ZERO) },
+            ];
+            for (const update of updates) {
+                // When trying to change the field
+                const result: Promise<void> = partDAO.update(partId, update);
+                // Then it should fail
+                await expectPermissionToBeDenied(result);
+            }
+        });
+        it('should forbid deleting part if owner has not timed out, even if they are not observing anymore', async() => {
```

Comment:
that's a non-started part !
So (dégueulasse ici mais sexy en monospace)
```
                        | Started | Non-Started
timed-out & observing   |    X    |      V
timed-in & observing    |    X    |      X
timed-in & nonbserving  |    X    |      X
timed-out & nonbserving |    X    |      V
```
8 tests si on prend ces trois facteur, naturellement, sinon ça peut éventuellement se résumer à ne pas mentionner observedPart qui n'intervient pas, et ne faire qu'un test pour started=neverremovable

Reply:
Fait 4 tests:

```
it('should forbid deleting non-started part if owner has not timed out', async() => { // timed in, non-started
it('should allow deleting non-started part if owner has timed out', async() => {  // timed out, non-started
it('should forbid deleting a started part if the owner has timed out', async() => { // timed out, started
it('should forbid deleting a started part if the owner has not timed out', async() => { // timed in, started
```